core/src/test/java/io/grpc/internal/ManagedChannelImplTest.java

```diff
@@ -2394,11 +2394,15 @@ public ClientStream answer(InvocationOnMock in) throws Throwable {
     when(mockPicker.pickSubchannel(any(PickSubchannelArgs.class)))
         .thenReturn(PickResult.withSubchannel(subchannel));
     updateBalancingStateSafely(helper, READY, mockPicker);
+    assertEquals(0, offloadFakeClock.numPendingTasks());
     executor.runDueTasks();
     ArgumentCaptor<RequestInfo> infoCaptor = ArgumentCaptor.forClass(null);
+    ArgumentCaptor<Executor> executorArgumentCaptor = ArgumentCaptor.forClass(null);
     ArgumentCaptor<CallCredentials.MetadataApplier> applierCaptor = ArgumentCaptor.forClass(null);
     verify(creds).applyRequestMetadata(infoCaptor.capture(),
-        same(executor.getScheduledExecutorService()), applierCaptor.capture());
+        executorArgumentCaptor.capture(), applierCaptor.capture());
+    assertTrue(((ManagedChannelImpl.ExecutorHolder)executorArgumentCaptor.getValue())
```

Comment:
`assertTrue()` produces a pretty useless error message when it fails. You should prefer an assertion that matches, as best you can. In this case, you can do `assertSame(offloadExecutor, ...getExecutor())` from JUnit. We also use Truth, so `assertThat(...getExecutor()).isSameInstanceAs(offloadExecutor)` would also be common. This is an old test, so it is using both so either one is fair game (there's no benefit one way or the other).